src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+
+
+#include "J2735MapToJsonConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    void J2735MapToJsonConverter::convertJ2735MAPToMapJSON(const std::shared_ptr<MapData> mapMsgPtr, Json::Value &mapJson) const
+    {
+        // Construct metadata
+        Json::Value metadata;
+        auto timestamp_utc = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        metadata["timestamp"] = std::to_string(timestamp_utc);
+        mapJson["metadata"] = metadata;
+
+        // Construct Map Data
+        Json::Value mapDataJson;
+        mapDataJson["layer_id"] = mapMsgPtr->layerID;
+        mapDataJson["msg_issue_revision"] = std::to_string(mapMsgPtr->msgIssueRevision);
+        mapDataJson["layer_type"] = std::to_string(*mapMsgPtr->layerType);
+
+        // Construct intersections
+        const IntersectionGeometryList *intersections = mapMsgPtr->intersections;
+        Json::Value intersectionsJson;
+        // Assume there is only one intersection geometry for each intersection
```

Comment:
Is this comment accurate. It looks like the for loop considers multiple intersections.

Reply:
Yeah, it is a for loop to check multiple intersection. But there is only one intersection geometry in this array. So I guess this is a note to remind of the number of intersection geometry during the testing with the west intersection MAP UPER file.